File: services/message-service/src/main/java/messenger/message/service/exception/mapper/ErrorMapper.java
package messenger.message.service.exception.mapper;

import mapper.AbstractErrorMapper;
import messenger.message.service.exception.AuthorizationException;
import messenger.message.service.exception.IllegalChatTypeException;
import messenger.message.service.exception.MessageNotFoundException;
import org.springframework.stereotype.Component;

@Component
public class ErrorMapper extends AbstractErrorMapper {

    @Override
    protected int getErrorCode(Throwable e) {
        if (e instanceof MessageNotFoundException) {
            return 404;
        } else if (e instanceof IllegalChatTypeException) {
            return 400;
        } else if (e instanceof AuthorizationException) {
            return 403;
        }

        return 500;
    }
}

File: services/media-service/src/main/kotlin/messenger/media/service/exception/mapper/ErrorMapper.kt
package messenger.media.service.exception.mapper

import messenger.media.service.dto.response.ErrorResponse
import messenger.media.service.exception.FileEmptyException
import messenger.media.service.exception.FileSizeException
import messenger.media.service.exception.MediaNotFoundException
import org.springframework.stereotype.Component
import java.time.Instant

@Component
class ErrorMapper {

    fun from(e: Exception): ErrorResponse {
        return ErrorResponse(
            errorCode = getErrorCode(e),
            error = e::class.simpleName,
            message = e.message,
            timestamp = Instant.now()
        )
    }

    private fun getErrorCode(e: Throwable): Int {
        return when (e) {
            is FileEmptyException -> 400
            is FileSizeException -> 400
            is MediaNotFoundException -> 404
            else -> 500
        }
    }
}